src/flask_pyoidc/flask_pyoidc.py

```diff
@@ -319,3 +342,100 @@ def valid_access_token(self, force_refresh=False):
+        '''RFC 7662: Token Introspection
+        The Token Introspection extension defines a mechanism for resource
+        servers to obtain information about access tokens. With this spec,
+        resource servers can check the validity of access tokens, and find out
+        other information such as which user and which scopes are associated
+        with the token.
+
+        Parameters
+        ----------
+        request : werkzeug.local.LocalProxy
+            flask request object
+        client : flask_pyoidc.pyoidc_facade.PyoidcFacade
+            PyoidcFacdae object contains metadata of the provider and client
+        scopes : list
+            Specify scopes that are bound for the end endpoint
+
+        Returns
+        -------
+        bool
+            True if access_token is valid
+        None
+            if access_token is invalid
+
+        Raises
+        ------
+        NotForMe
+            if access_token is valid
+            
+        How To Use?
+        -----------
+        Enable token introspection globally by provding access_token_required to
+        True as an argument.
+        >>>> auth = OIDCAuthentication({'default': provider_config},
+                                       access_token_required=True)
+
+        >>>> # You can also enable or disable it per API endpoint from the decorator by
+        # providing accept_token to either True or False as an argument. It has
+        # higher precedence than global argument.
+
+        >>>> @auth.oidc_auth('default', accept_token=True)
+        >>>> # Additionally, you can provide required scopes for your endpoint.
+        >>>> @auth.oidc_auth('default', accept_token=True, scopes=['read', 'write'])
+        '''
+        received_access_token = self._parse_access_token(request)
+        # send token introspection request
+        result = client._token_introspection_request(
+            access_token=received_access_token)
+        logger.debug(result)
+        # Check if access_token is valid, active can be True or False
+        if result.get('active'):
+            # Check if client_id is in audience claim
+            if not client._client.client_id in result['aud']:
+                logger.info('Token is valid but required audience is missing')
+                # raises exception if client_id is not in audience, you can
+                # configure audience from Identity Provider
+                raise NotForMe('required audience is missing')
+            # Check if scopes associated with the access_token are the ones
+            # given by the user and not something else which is not permitted.
+            if scopes is not None:
+                if not set(scopes).issubset(set(result['scope'].split())):
+                    logger.info('Token is valid but does not have required scopes')
+                    raise NotForMe('Token does not have required scopes')
+            return True
```

Comment:
* Missing return for the negative case (no `return False`).
  The deep nesting could also be avoided by reversing the statements, i.e.:

   ```python
   if not result.get('active'):
     return False

   if not client._client.client_id in result['aud']:
     logger.info('Token is valid but required audience is missing')
     raise NotForMe('required audience is missing')

   etc...
   ```
* As mentioned in a comment above, some of the data in the Introspection response should be passed to the view function. Hence this function needs to return either the IntrospectionResponse or a subset of its data instead of a boolean.